src/lib/common/win32/KnownFolderPaths.h

```diff
@@ -0,0 +1,32 @@
+/*
+* barrier -- mouse and keyboard sharing utility
+* Copyright (C) 2018 Debauchee Open Source Group
+*
+* This package is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public License
+* found in the file LICENSE that should have accompanied this file.
+*
+* This package is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include <string>
+
+class KnownFolderPaths
+{
+public:
+    static std::string desktop();
```

Comment:
I think it makes sense to have these as simple functions.

Reply:
Pushed an update, but not sure if it was something like this you had in mind, or something else? (and if so, I'm happy to make another attempt)

Reply:
Pushed another update, because I managed to track down another one of those codepage issues. Refactored the KnowFolderPaths into a header-only, and added another one, StringConverters.h, for windows string conversions (narrow ansi, narrow utf8, and wide). But again, just let me know if you think it would be better doing it differently!